port/win/env_win.cc

```diff
@@ -436,19 +436,136 @@ Status WinEnvIO::GetFileModificationTime(const std::string& fname,
   return s;
 }
 
+static BOOL DeleteDirRecursively(const char* dir) 
+{
+    HANDLE hFind;
+    WIN32_FIND_DATA findFileData;
+
+    char path[MAX_PATH];
+    char name[MAX_PATH];
+
+    strcpy(path, dir);
+    strcat(path, "\\*");
+    strcpy(name, dir);
+    strcat(name, "\\");
+
+    hFind = FindFirstFileA(path, &findFileData);
+    if (hFind == INVALID_HANDLE_VALUE) 
+        return FALSE;
+
+    strcpy(path, name);
+    while (true)
+    {
+        if (FindNextFileA(hFind, &findFileData)) 
+        {
+            if (strcmp(findFileData.cFileName, ".") == 0
+                || strcmp(findFileData.cFileName, "..") == 0)
+                continue;
+
+            strcat(name, findFileData.cFileName);
+
+            // if dir
+            if ((findFileData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)) 
+            {
+                if (!DeleteDirRecursively(name) || !RemoveDirectoryA(name) )
+                {
+                    auto err = ::GetLastError();
+                    FindClose(hFind);
+                    ::SetLastError(err);
+                    return FALSE;
+                }
+            }
+
+            // else file
+            else 
+            {
+                if (findFileData.dwFileAttributes & FILE_ATTRIBUTE_READONLY)
+                    _chmod(name, _S_IWRITE); // change read-only file mode
+                
+                if (!DeleteFileA(name)) 
+                {
+                    auto err = ::GetLastError();
+                    FindClose(hFind);
+                    ::SetLastError(err);
+                    return FALSE;
+                }
+            }
+
+            strcpy(name, path);
+        }
+        else
+        {
+            auto err = ::GetLastError();
+            if (err == ERROR_NO_MORE_FILES)
+                break;
+            else 
+            {
+                FindClose(hFind);
+                ::SetLastError(err);
+                return FALSE;
+            }
+        }
+
+    }
+
+    FindClose(hFind);
+    return RemoveDirectoryA(dir);
+}
+
 Status WinEnvIO::RenameFile(const std::string& src,
```

Comment:
I assume that you need this new feature for to write code on its basis. If so you want to make sure that this is also implemented on other platforms otherwise your code will not run there.

Alternatively, you can come up with a new interface and make sure it returns not implemented on other platforms so client code can fallback.

If, on the other hand, you want to create a private build you can inherit from this class and override just this function and come up with your own env for your purposes.

Log rotation feature relies on this so you want to make sure that all standalone tests pass.